pkg/utils/kubernetes/client/client.go

```diff
@@ -430,3 +432,21 @@ func ApplyToObjectKinds(ctx context.Context, fn func(kind string, objectList cli
 
 	return flow.Parallel(taskFns...)(ctx)
 }
+
+// ForceDeleteObjects lists and finalizes all the objects in the passed namespace and deletes them.
+func ForceDeleteObjects(ctx context.Context, log logr.Logger, c client.Client, kind string, namespace string, objectList client.ObjectList) flow.TaskFn {
```

Comment:
Did you check whether this function could also be used in the logic of the migration flow?

Reply:
In the `ForceDeleteObjects`, we forcefully finalize the object. Do we want to do the same for migration? I couldn't find such occurrences in the migration flow.